knative-operator/pkg/common/util.go

```diff
@@ -40,21 +41,19 @@ func BuildImageOverrideMapFromEnviron(environ []string) map[string]string {
 
 	for _, e := range environ {
 		pair := strings.SplitN(e, "=", 2)
-		if strings.HasPrefix(pair[0], "IMAGE_") {
+		if strings.HasPrefix(pair[0], ImagePrefix) {
 			// convert
 			// "IMAGE_container=docker.io/foo"
-			// "IMAGE_deployment_container=docker.io/foo2"
+			// "IMAGE_deployment__container=docker.io/foo2"
+			// "IMAGE_env_var=docker.io/foo3"
+			// "IMAGE_deployment__env_var=docker.io/foo4"
 			// to
 			// container: docker.io/foo
 			// deployment/container: docker.io/foo2
-			var name string
-			parts := strings.SplitN(pair[0], "_", -1)
```

Comment:
If we had set this to `2`, would we theoretically get the same behavior even with underscores?

Regardless: I prefer a "proper" sentinel, double underscores seem fine.

Reply:
> If we had set this to 2, would we theoretically get the same behavior even with underscores?

`IMAGE_PING_IMAGE` might not work, depending on how you implement it.

That's the problematic case.
We receive the env var as `IMAGE_PING_IMAGE=docker.io/foo` and deployment name is not `PING` :)

> Regardless: I prefer a "proper" sentinel, double underscores seem fine.

Yeah, me too. Also makes human-reading of CSV easier.

Reply:
Gotcha, yeah we need a proper sentinel then.